dev/benchmarks/macrobenchmarks/test/util.dart

```diff
@@ -57,12 +59,42 @@ void macroPerfTestE2E(
       await setup(tester);
     }
 
-    await binding.watchPerformance(() async {
+    Future<void> testBody() async {
       final Future<void> durationFuture = tester.binding.delayed(duration);
       if (body != null) {
         await body(tester);
       }
       await durationFuture;
+    }
+
+    vm.Timeline timeline;
+    await binding.watchPerformance(() async {
+      if (measureCpuGpu) {
+        timeline = await binding.traceTimeline(testBody, streams: <String>['Embedder']);
+      } else {
+        await testBody();
+      }
     });
+
+    if (measureCpuGpu) {
+      final List<double> cpuUsage = timeline.traceEvents.where(
```

Comment:
Can we reuse [ProfilingSummarizer][1] for the computation below?

[1]: https://github.com/flutter/flutter/blob/master/packages/flutter_driver/lib/src/driver/profiling_summarizer.dart

Reply:
Similar to the reason we need a `FrameTimingSummarizer` but basically duplicating `TimelineSummary`: `ProfilingSummarizer` is very coupled to `TimelineEvent` in the `flutter_driver`,  which is similar but a different class to `Timeline` in `vm_service`. If I want to use `ProfilingSummarizer` I will need to 1) introduce the whole `flutter_driver` dependence to the test app, which I'm not sure if there will be issue because the related code is written for host machine; 2) either write a very large converter from `Timeline` in `vm_service` to `flutter_driver`'s `TimelineSummary` or to serialize it to json and deserialize it back to `flutter_driver`.

Reply:
I see. Alternatively, can we move `flutter_driver`'s `Timeline, TimelineEvent, ProfilingSummarizer` to `integration_test` package, and let `flutter_driver` use `integration_test`? All those 3 classes don't seem to depend on anything extra from `flutter_driver`. Serializing vm_service's Timeline to json and deserialize it back to the specialized Timeline seems to be fine and not a lot of work.

CC @iskakaushik : as the author of `ProfilingSummarizer`, do you see any concern of moving `ProfilingSummarizer` to `integration_test` so we can use it in more places?

Reply:
If we will be able to re-structure these codes, I would say that `Timeline`, `TimelineEvent` in `flutter_driver` and in `vm_service` look like something we can unify to the same class/interface, i.e. use/inherit the `vm_service` version in `flutter_driver`. cc @iskakaushik let me know if there's anything that prevent this.

Reply:
And as we already have `FrameTimingSummarizer` in `flutter_test`, if `integration_test` is not a good place to hold these code, `flutter_test` can also be an alternative.